rholang/src/test/scala/coop/rchain/rholang/interpreter/accounting/CostAccountingSpec.scala

```diff
@@ -63,4 +72,98 @@ class CostAccountingSpec extends FlatSpec with TripleEqualsSupport {
 
     assertOutOfPhloError(test)
   }
+
+}
+
+class CostAccountingSpec extends FlatSpec with Matchers {
+
+  private[this] def evaluateWithCostLog(
+      initialPhlo: Long,
+      contract: String
+  ) = {
+    implicit val errorLog = new ErrorLog[Task]()
+    implicit val costAlg: CostAccounting[Task] =
+      CostAccounting.unsafe[Task](Cost(initialPhlo))
+    implicit val costL             = costLog[Task]
+    implicit val cost: _cost[Task] = loggingCost(costAlg, costL)
+    val dbDir                      = Files.createTempDirectory("cost-accounting-spec-")
+    val size                       = 1024L * 1024 * 1024
+    implicit val logF: Log[Task]   = new Log.NOPLog[Task]
+
+    (for {
+      costsLoggingProgram <- costL.listen({
+                              for {
+                                runtime <- Runtime
+                                            .create[Task, Task.Par](dbDir, size, StoreType.LMDB)
+                                par <- Interpreter[Task].buildNormalizedTerm(contract)
+                                res <- Interpreter[Task]
+                                        .evaluate(runtime, par, Cost(initialPhlo.toLong))
+                                _ <- runtime.close()
+                                _ <- Task.eval(dbDir.recursivelyDelete())
+                              } yield (res)
+                            }.attempt)
+      (result, costLog) = costsLoggingProgram
+      res               <- errorLog.readAndClearErrorVector
+      _                 = Task.now(res should be(Vector.empty))
```

Comment:
😈 This should be either `_  <- Task.now(res should be(Vector.empty))` or `_  = res should be(Vector.empty)`. As it stands, the assertion is never evaluated.

Reply:
Good catch! Cheers

Reply:
Actually, when I change the assertion to fail `      _                 = Task.now(res should not be (Vector.empty))` it does fail:
```
[info] Running out of phlogistons
[info] - should stop the evaluation upon cost depletion in a single execution branch *** FAILED *** (2 seconds, 572 milliseconds)
[info]   Vector() was equal to Vector() (CostAccountingSpec.scala:117)
```